runtime/wasm/src/host_exports.rs

```diff
@@ -630,6 +636,14 @@ impl HostExports {
             panic!("Critical error logged in mapping");
         }
     }
+
+    pub(crate) fn data_source_address(&self) -> H160 {
+        self.data_source_address.clone().unwrap_or_default()
+    }
+
+    pub(crate) fn data_source_network(&self) -> String {
+        self.data_source_network.clone()
```

Comment:
And shouldn't these two methods return `Option<_>` as well to map to `Address | null` / `string | null` in AssemblyScript?

Reply:
The reasoning to represent `None` as an empty string is that data sources with an address (and network) are the common case, and if you don't have an address you wouldn't be calling this export anyways, so I tried to save users the hassle of checking null and casting to `String`.

Reply:
While I think that's reasonable, I could see users writing helper functions used in multiple data sources where they _don't_ have a priori knowledge of whether there is an address / network.

Then again, in those cases checking for `''` is no worse than checking for `null`.